synapse/api/auth.py

```diff
@@ -367,6 +373,31 @@ def is_membership_change_allowed(self, event, auth_events):
 
         return True
 
+    def _verify_third_party_invite(self, event, auth_events):
```

Comment:
Could we have a quick docstring/comment here to outline what this actually checks?